dashboard/app/helpers/levels_helper.rb

```diff
@@ -196,6 +196,9 @@ def app_options
       @app_options = view_options.camelize_keys
     end
 
+    # Blockly caches level properties, whereas this field depends on the user
+    @app_options['teacherMarkdown'] = @level.properties['teacher_markdown'] if @user.try(:authorized_teacher?)
```

Comment:
surprisingly, it turns out @user is actually different than current_user here (maybe related to pair programming) and current_user is actually what we want. Caught while going to write an eyes test :)